tools/cli/src/mentat_cli/repl.rs

```diff
@@ -7,40 +7,88 @@
 // under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
 // CONDITIONS OF ANY KIND, either express or implied. See the License for the
 // specific language governing permissions and limitations under the License.
+
+use store::Store;
 use input::{InputReader};
-use input::InputResult::{Command, Empty, More, Eof};
+use input::InputResult::{Command, Empty, More, Eof, InputError};
 
 /// Starting prompt
 const DEFAULT_PROMPT: &'static str = "mentat=> ";
 /// Prompt when further input is being read
+// TODO: Should this actually reflect the current open brace?
 const MORE_PROMPT: &'static str = "mentat.> ";
-/// Prompt when a `.block` command is in effect
-const BLOCK_PROMPT: &'static str = "mentat+> ";
+
+/// Describes what type of arguments, if any, a command may accept.
+#[derive(Copy, Clone, Debug, Eq, PartialEq)]
+pub enum CmdArgs {
+    /// No arguments
+    Nothing,
+    /// Command accepts a local filename
+    Filename,
+    /// Optional unprocessed text may be accepted
+    Text,
+}
+
+/// Represents a REPL command definition
+#[derive(Debug)]
+pub struct CommandDef {
+    pub name: &'static str,
+    pub args: Option<&'static str>,
+    pub accepts: CmdArgs,
+    pub help: &'static str,
+}
+
+/// List of commands
+static COMMANDS: &'static [CommandDef] = &[
+    CommandDef{name: "open", args: Some("<database>"),
+        accepts: CmdArgs::Text,
+        help: "Open a database with the provided name. Creates a new database of that name if it doesn't already exist."},
+    CommandDef{name: "help", args: Some("[command]"),
+        accepts: CmdArgs::Text,
+        help: "Show help for commands"},
+];
+
+/// Looks up a command name by what may be an abbreviated prefix.
+/// Returns the `CommandDef` structure if one is found.
```

Comment:
This is a hacky way of doing what getopts calls abbreviation to uniqueness. It'll only work so long as none of your commands share a prefix — `close` and `create`, for example, will clash for `.c`.

Think about whether you need to do this.